plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/camera/MapAnimator.java

```diff
@@ -0,0 +1,160 @@
+      // Get the current target from the current map camera position
+      currentPosition = mapboxMap.getCameraPosition();
+    }
+
+    public Builder addLatLngAnimator(@NonNull LatLngAnimator latLngAnimator) {
+
+      LatLng currentTarget = currentPosition.target;
+      if (currentTarget == latLngAnimator.getTarget()) {
+        return this;
+      }
+
+      latLngAnimator.setObjectValues(currentTarget, latLngAnimator.getTarget());
+      latLngAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+        @Override
+        public void onAnimationUpdate(ValueAnimator animation) {
+          mapboxMap.setLatLng((LatLng) animation.getAnimatedValue());
+        }
+      });
+
+      animators.add(latLngAnimator);
+      return this;
+    }
+
+    public Builder addZoomAnimator(@NonNull ZoomAnimator zoomAnimator) {
+
+      float currentZoom = (float) currentPosition.zoom;
+      if (currentZoom == zoomAnimator.getTargetZoom()) {
+        return this;
+      }
+
+      zoomAnimator.setFloatValues(currentZoom, zoomAnimator.getTargetZoom());
+      zoomAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+        @Override
+        public void onAnimationUpdate(ValueAnimator animation) {
+          mapboxMap.setZoom((Float) animation.getAnimatedValue());
+        }
+      });
+
+      animators.add(zoomAnimator);
+      return this;
+    }
+
+    public Builder addBearingAnimator(@NonNull BearingAnimator bearingAnimator) {
+
+      float currentBearing = (float) currentPosition.bearing;
+      float normalizedTargetBearing = normalizeBearing(currentBearing, bearingAnimator.getTargetBearing());
+      if (currentBearing == normalizedTargetBearing) {
+        return this;
+      }
+
+      bearingAnimator.setFloatValues(currentBearing, normalizedTargetBearing);
+      bearingAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+        @Override
+        public void onAnimationUpdate(ValueAnimator animation) {
+          mapboxMap.setBearing((Float) animation.getAnimatedValue());
+        }
+      });
+
+      animators.add(bearingAnimator);
+      return this;
+    }
+
+    public Builder addTiltAnimator(@NonNull TiltAnimator tiltAnimator) {
+
+      float currentTilt = (float) currentPosition.tilt;
+      if (currentTilt == tiltAnimator.getTargetTilt()) {
+        return this;
+      }
+
+      tiltAnimator.setFloatValues(currentTilt, tiltAnimator.getTargetTilt());
+      tiltAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+        @Override
+        public void onAnimationUpdate(ValueAnimator animation) {
+          mapboxMap.setTilt((Float) animation.getAnimatedValue());
+        }
+      });
+
+      animators.add(tiltAnimator);
+      return this;
+    }
+
+
+
+    public MapAnimator build() {
+      return new MapAnimator(animators);
+    }
+  }
+
+  private static float normalizeBearing(float currentBearing, float targetBearing) {
```

Comment:
We should just extract this method into the Utils class since I already use the same logic in `LocationLayerPlugin`

Reply:
@cammace Sure thing.  Just porting this code from https://github.com/mapbox/mapbox-navigation-android/pull/679 - will look to clean up and remove duplicate code as the PR progresses

Reply:
awesome